hasura/graphql-engine
# log when request body parsing fails (fix #2555)

### Description


Regression: since `beta.3` whenever request body parsing failed, the error was not logged. This PR fixes the bug.

### Affected components 


- Server

### Related Issues



#2555 

### Solution and Design


Simple fix. The `parseBody` function was previously part of the handler functions. Since `beta.3` it has been moved to `mkSpockAction`, but the result of it was handled by `qErrToResp` instead of `logAndThrow` function.

**Edit**: After review from @rakeshkky , refactored `mkSpockAction` to have separate functions for [logging error and returning the response] and [logging success and returning the response]. Removed specific functions which only returned response (which was the source of the bug).

### Notes to reviewer
- ~~Simple one line fix in `Hasura.Server.App` module~~


### Steps to test and verify


1. Run graphql-engine
2. Make a request to `/v1/graphql` or `/v1/query` with wrong JSON structure.
3. Observe: The error is logged

### Limitations, known bugs & workarounds
